From 93a78e891a15adc9d674d92c3897d61706c870c2 Mon Sep 17 00:00:00 2001
From: 24bit192khz <118996408+24bit192kHz@users.noreply.github.com>
Date: Wed, 30 Sep 2026 17:23:18 +0300
Subject: [PATCH 24/24] mb/lenovo/sklkbl_thinkpad/t480: Radios on, no tablet
 switch, lid wakes from S4, GMM, subsystem IDs

As on the vendor firmware: the radios are never hard-blocked, no tablet
mode switch, the lid wakes from S4, the GMM device 00:08.0 is on, and all
PCH and system agent devices carry 17aa:225d.
---
 src/mainboard/lenovo/sklkbl_thinkpad/Kconfig              | 8 ++++++++
 src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl          | 1 +
 .../lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb  | 1 +
 3 files changed, 10 insertions(+)

diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
index 90d1965d66..ca15468a6b 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/Kconfig
@@ -14,6 +14,8 @@ config BOARD_LENOVO_SKLKBL_THINKPAD_COMMON
 	select H8_HAS_FNLOCK_LED
 	select H8_HAS_PRIMARY_FN_KEYS
 	select H8_SUPPORT_BT_ON_WIFI
+	select H8_NO_WIRELESS_SWITCH
+	select H8_NO_TABLET_MODE_SWITCH
 	select HAVE_ACPI_RESUME
 	select HAVE_ACPI_TABLES
 	select INTEL_GMA_HAVE_VBT
@@ -128,3 +130,9 @@ config VARIANT_HAS_DGPU
 	bool
 
 endif
+
+config SUBSYSTEM_VENDOR_ID
+	default 0x17aa if BOARD_LENOVO_T480
+
+config SUBSYSTEM_DEVICE_ID
+	default 0x225d if BOARD_LENOVO_T480
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl b/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
index 3a949a2fca..4dc31bf982 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/acpi/ec.asl
@@ -3,6 +3,7 @@
 #define BRIGHTNESS_UP \_SB.PCI0.GFX0.INCB
 #define BRIGHTNESS_DOWN \_SB.PCI0.GFX0.DECB
 #define THINKPAD_EC_GPE 22
+#define H8_LID_WAKE_STATE 4	/* as the vendor firmware */
 
 Name(\TCRT, 100)
 Name(\TPSV, 90)
diff --git a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
index c9c429ae1a..4eaf6da3c4 100644
--- a/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
+++ b/src/mainboard/lenovo/sklkbl_thinkpad/variants/t480/overridetree.cb
@@ -10,6 +10,7 @@ chip soc/intel/skylake
 	}"
 
 	device domain 0 on
+		device ref gmm on end	# Gaussian Mixture Model accelerator, as the vendor firmware
 		device ref south_xhci on
 			register "usb2_ports" = "{
 				[0] = USB2_PORT_MID(OC1),		// USB-A
